### DISCUSSION TITLE
Updates Set methods in ScalarFields

### DISCUSSION THREAD

**Discussion by shishiousan** [easifem/classes#241] [Ideas]
I think this issue is strongly related the issue number #179  

When a `ScalarField_` is set by the `addContribution=.FALSE.` and given `scale` value, it does not work properly.

For example:
 
```fortran 
CALL obj%set(VALUE=avec,  &
      & addContribution=.FALSE., &
      & scale=alp)
```

currently, performs following operations. 

```txt
obj = obj + alp * avec
```  

But, it should perform following:

```txt
obj = alp*avec
```

**Comment by vickysharma0812** [comment]
`ScalarField` (and other fields) under the hood use routines from base library (see `RealVector_SetMethods` module).

As @shishiousan  San suggested, we can make an internal copy called `alpha * value` and call the current `Set` method. But, I do not recommend this,  because we have to repeat this for other fields, in addition, we make additional copy to store `alpha * value`.

Therefore, we should add optional argument in the `Set` method defined in `RealVector_SetMethods`.

Regards
Vikas
